cdap-ui/app/hydrator/controllers/create/toppanel-ctrl.js

```diff
@@ -66,14 +67,32 @@ class HydratorPlusPlusTopPanelCtrl {
                             MY_CONFIG.hydrator.previewEnabled === true;
 
     this.previewMode = false;
-    this.previewStartTime = null;
-    this.displayDuration = {
-      minutes: '--',
-      seconds: '--'
-    };
-    this.previewTimerInterval = null;
-    this.previewLoading = false;
-    this.previewRunning = false;
+    this.previewLoading = true;
+
+    if (this.currentPreviewId) {
+      this.myPreviewLogsApi.getLogsStatus({
+        namespace : this.$state.params.namespace,
+        previewId : this.currentPreviewId
+      }).$promise.then(
+        (statusRes) => {
+          if (statusRes.status === 'RUNNING') {
+            this.previewStartTime = statusRes.startTime;
+            this.previewLoading = false;
+            this.previewRunning = true;
+            this.startTimer();
+            this.startPollPreviewStatus(this.currentPreviewId);
+          } else {
+            this.setDefault();
+          }
+        },
+        (statusErr) => {
+          console.log('ERROR: ', statusErr);
+          this.setDefault();
```

Comment:
Added commits to show both duration and logs of previous preview. As soon as the user restarts the preview, both will be cleared out.